src/test/java/seedu/address/model/fee/MonthlyFeeTest.java

```diff
@@ -0,0 +1,47 @@
+package seedu.address.model.fee;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class MonthlyFeeTest {
+
+    private double validMonthlyFee = 30.20;
+    private Month validMonth = new Month(3);
+    private Year validYear = new Year(2021);
+
+    @Test
+    public void constructor_invalidMonthlyFee_throwsNullPointerException() {
+        assertThrows(java.lang.NullPointerException.class, () -> new MonthlyFee(validMonthlyFee, validMonth, null));
+        assertThrows(java.lang.NullPointerException.class, () -> new MonthlyFee(validMonthlyFee, null, validYear));
```

Comment:
Would it make sense to check for **negative** monthly fee here, and as well as in your `MonthlyFee.java`?

Reply:
O.o right! But its an illegal exception error instead, did that in the latest commit.